server/svix-server/src/queue/rabbitmq.rs

```diff
@@ -0,0 +1,269 @@
+use chrono::Utc;
+use futures::StreamExt;
+use lapin::{
+    options::{
+        BasicConsumeOptions, BasicPublishOptions, BasicQosOptions, ExchangeDeclareOptions,
+        QueueBindOptions, QueueDeclareOptions,
+    },
+    types::{AMQPValue, FieldTable},
+    BasicProperties, ConnectionProperties,
+};
+use svix_ksuid::{KsuidLike, KsuidMs};
+
+use crate::{ctx, err_generic, err_queue, error::Result};
+use std::{sync::Arc, time::Duration};
+
+use super::{
+    Acker, QueueTask, TaskQueueConsumer, TaskQueueDelivery, TaskQueueProducer, TaskQueueReceive,
+    TaskQueueSend,
+};
+
+#[derive(Clone)]
+pub struct Producer(Arc<ProducerInner>);
+
+struct ProducerInner {
+    exchange_name: String,
+    queue_name: String,
+    channel: lapin::Channel,
+}
+
+pub struct Consumer {
+    consumer: lapin::Consumer,
+}
+
+pub async fn new_pair(
+    dsn: &str,
+    queue_name: String,
+) -> Result<(TaskQueueProducer, TaskQueueConsumer)> {
+    let conn = ctx!(lapin::Connection::connect(dsn, ConnectionProperties::default()).await)?;
+    let producer_chan = ctx!(conn.create_channel().await)?;
+    let consumer_chan = ctx!(conn.create_channel().await)?;
```

Comment:
I'm not sure how this fits with the abstraction, but it's not uncommon to have the connection break, requiring some sort of reconnect loop logic. I wonder where this can happen as written. If the pipe breaks, what do? Does that require the process to restart?

Reply:
We discussed this a bit offline, so adding those notes here for clarity

Handling broken connections is definitely a prerequisite for using this in prod. The main concerns are
1. Since `lapin` doesn't yet handle this for us (maybe [one day](https://github.com/amqp-rs/lapin/issues/312)), there's some hefty synchronization we'd have to add to handle graceful reconnections _given how our work queue is currently set up_. i.e. This is very complicated
2. We still need more time to exercise RabbitMQ and discover all the kinks, so we wouldn't want to recommend folks jump into using RabbitMQ for production projects yet _anyway_

Therefore, we're gonna table the reconnection issue for another PR.

Reply:
Added https://github.com/svix/svix-webhooks/issues/898